### DISCUSSION TITLE
Is there way for a block to know if it's in the block editor, the widget editor, or the customizer-widget-editor?

### DISCUSSION THREAD

**Discussion by jakeparis** [WordPress/gutenberg#46889] [Q&A]
I was able to write something for a block to know whether it's being displayed in the block editor or the widget editor (admin page)... basically by checking if the block has a "core/widget-area" as a parent/ancestor. 

```js
import { useSelect, select } from '@wordpress/data';

...

const parentIds = useSelect( select => select('core/block-editor').getBlockParents( props.clientId ) );

var inWidgetArea = false;
if( parentIds ) {
    parentIds.forEach( parentId => {
	  let parent = select('core/block-editor').getBlock( parentId );
	  if( parent.name === 'core/widget-area' )
		  inWidgetArea = true;
    });
}
```

However, the customizer version of the widget editor doesn't wrap the blocks in a `core/widget-area` parent block. Anyone know of a different way to know which of these three contexts my block is displaying within?

**Comment by devfle** [comment]
You could try getting the screen instance from the respective admin page and compare it. If you need the info in the frontend, you can pass it through.

```php
add_action('current_screen', function() {
    $screen = get_current_screen();

    if ($screen->base === 'customize') echo 'Customizer Screen';
}, 10);
```
